go/libraries/doltcore/sqle/database.go

```diff
@@ -44,62 +47,129 @@ const (
 	single  batchMode = false
 )
 
+type tableCache struct {
+	mu *sync.Mutex
+	tables map[string]map[*doltdb.RootValue]map[string]sql.Table
+}
```

Comment:
I'm still not convinced that a per root cache is going to give the correct behavior in the long term, but maintaining it for now.